dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/test/com/ibm/ws/jaxrs/test/MessageImplTest.java

```diff
@@ -0,0 +1,525 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jaxrs.test;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.AbstractMap;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.cxf.message.Message;
+import org.apache.cxf.message.MessageImpl;
+import org.apache.cxf.transport.Conduit;
+import org.apache.cxf.transport.Destination;
+import org.apache.cxf.transport.MessageObserver;
+import org.apache.cxf.ws.addressing.EndpointReferenceType;
+import org.junit.Test;
+
+import junit.framework.Assert;
+
+
+public class MessageImplTest {
+    
+    @Test
+    public void testMessageImpl() throws Exception {
+        MessageImpl message = new MessageImpl();
+        String[] propertyNames = message.getPropertyNames();
```

Comment:
I was hoping that you wouldn't need to have the getGetters, getSetters, getRemoves, getContains in MessageImpl.  You can build that dynamically using reflection and it will be less fragile and the test can automatically pick up new getters, etc and fail tests when new methods are added.